src/tools/std20/views/test/counted.test.cpp

```diff
@@ -0,0 +1,207 @@
+//
+//    WHEN( "when iterators are used" ) {
+//
+//      auto chunk = std20::views::counted( values.begin(), 3 );
+//      using Range = decltype(chunk);
+//      using Iterator = nano::iterator_t< Range >;
+//
+//      THEN( "the take_view satisfies the required concepts" ) {
+//
+//		    CHECK( std20::ranges::view< Range > );
+//		    CHECK( std20::ranges::sized_range< Range > );
+//        CHECK( std20::ranges::forward_range< Range > );
+//        CHECK( std20::ranges::bidirectional_range< Range > );
+//		    CHECK( ! std20::ranges::random_access_range< Range > );
+//		    CHECK( ! std20::ranges::contiguous_range< Range > );
+//		    CHECK( ! std20::ranges::common_range< Range > );
+//      }
+//
+//      THEN( "the take_view range and iterator associated types are correct" ) {
+//
+//        CHECK( std20::same_as< nano::ranges::range_value_t< Range >, int > );
+//        CHECK( std20::same_as< nano::ranges::range_reference_t< Range >, int& > );
+//        CHECK( std20::same_as< nano::ranges::range_difference_t< Range >, std::ptrdiff_t > );
+//
+//        CHECK( std20::same_as< nano::ranges::iter_value_t< Iterator >, int > );
+//        CHECK( std20::same_as< nano::ranges::iter_reference_t< Iterator >, int& > );
+//        CHECK( std20::same_as< nano::ranges::iter_difference_t< Iterator >, std::ptrdiff_t > );
+//      }
+//
+//      THEN( "an take_view can be constructed and members can be tested" ) {
+//
+//        CHECK( 3 == chunk.size() );
+//        CHECK( false == chunk.empty() );
+//        CHECK( true == bool( chunk ) );
+//
+//        CHECK( std20::ranges::equal( chunk, equal ) );
+//
+//        // the following should not compile: no random access iterator
+//        // CHECK( -2 == chunk[0] );
+//
+//        unsigned int counter = 0;
+//        for ( auto&& value : chunk ) {
+//
+//          value += counter;
+//          ++counter;
+//        }
+//        CHECK( 3 == counter );
+//
+//        CHECK( std20::ranges::equal( chunk, modified ) );
+//        CHECK( std20::ranges::equal( values, original_modified ) );
+//      } // THEN
+//    } // WHEN
+//  } // GIVEN
+
+  GIVEN( "a container with random access iterators" ) {
+
+    std::vector< int > values = { -2, -1, 0, 1, 2 };
+
+    WHEN( "when iterators are used" ) {
+
+      auto chunk = std20::views::counted( values.begin(), 3 );
+      using Range = decltype(chunk);
+      using Iterator = nano::iterator_t< Range >;
+
+      THEN( "the take_view satisfies the required concepts" ) {
+
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( std20::ranges::sized_range< Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( std20::ranges::bidirectional_range< Range > );
+		    CHECK( std20::ranges::random_access_range< Range > );
+		    CHECK( ! std20::ranges::contiguous_range< Range > );
+		    CHECK( std20::ranges::common_range< Range > );
+      }
+
+      THEN( "the take_view range and iterator associated types are correct" ) {
```

Comment:
"take_view" -> "counted" 

Likewise for other THENs in scenario

Reply:
same